fileshare_process/grpc_fileshare_process.go

```diff
@@ -0,0 +1,172 @@
+package fileshare_process
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/NordSecurity/nordvpn-linux/fileshare/pb"
+	"github.com/NordSecurity/nordvpn-linux/internal"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/status"
+)
+
+type StartupErrorCode int
+
+const (
```

Comment:
Will this errors need to be handled also by the GUI app?

Reply:
Some will probably need to be, at least to display correct error messages to the user. Depends on the GUI implementation.